tower-balance/src/choose/mod.rs

```diff
@@ -13,12 +12,12 @@ pub trait Choose<K, N> {
     ///
     /// `replicas` cannot be empty, so this function must always return a valid index on
     /// [0, replicas.len()-1].
-    fn choose(&mut self, replicas: Replicas<K, N>) -> usize;
+    fn choose(&mut self, replicas: Replicas<'_, K, N>) -> usize;
 }
 
 /// Creates a `Replicas` if there are two or more services.
 ///
-pub(crate) fn replicas<K, S>(inner: &IndexMap<K, S>) -> Result<Replicas<K, S>, TooFew> {
+pub(crate) fn replicas<K, S>(inner: &IndexMap<K, S>) -> Result<Replicas<'_, K, S>, TooFew> {
```

Comment:
It feels wonky to have this here, but not on `& '_ IndexMap<...>`

Reply:
In all other places, rustc is able to infer the lifetimes. I can add more explicit lifetime annotations like:

```rust
pub(crate) fn replicas<K, S>(inner: &'_ IndexMap<K, S>) -> Result<Replicas<'_, K, S>, TooFew>
```

...and a lifetime parameter to the `Choose` trait as such:

```rust
pub trait Choose<K, N> {
    /// Returns the index of a replica to be used next.
    ///
    /// `replicas` cannot be empty, so this function must always return a valid index on
    /// [0, replicas.len()-1].
    fn choose(&mut self, replicas: Replicas<'_, K, N>) -> usize;
}
```

...but we don't need those explicit annotations in those locations.